src/Middleware/auth/auth.guard.ts

```diff
@@ -2,56 +2,45 @@ import { Injectable, NestMiddleware } from '@nestjs/common';
 import { Request, Response, NextFunction } from 'express';
 import * as firebase from 'firebase-admin';
 import { FirebaseApp } from '../../Firebase/firebase.service';
-// import * as serviceAccount from '../../Firebase/firebaseConfig.json';
+import { auth } from 'firebase-admin';
+import UserRecord = auth.UserRecord;
 
-// const firebase_params: object = {
-//   type: serviceAccount.type,
-//   project_id: serviceAccount.project_id,
-//   private_key_id: serviceAccount.private_key_id,
-//   private_key: serviceAccount.private_key,
-//   client_email: serviceAccount.client_email,
-//   client_id: serviceAccount.client_id,
-//   auth_uri: serviceAccount.auth_uri,
-//   token_uri: serviceAccount.token_uri,
-//   auth_provider_x509_cert_url: serviceAccount.client_x509_cert_url,
-//   client_x509_cert_url: serviceAccount.client_x509_cert_url,
-//   universe_domain: serviceAccount.universe_domain,
-// };
 @Injectable()
 export class PreauthMiddleware implements NestMiddleware {
```

Comment:
Not clear is it a auth guard or preauth middleware?

Reply:
I have removed this middleware, and now my authentication is handled by a guard.